### DISCUSSION TITLE
OAUTH2 Laravel passport

### DISCUSSION THREAD

**Discussion by AbrorGOF** [saloonphp/saloon#109] [Q&A] [answered]
**Hi everyone, can I do like this code in saloon oauth 2**
```
       $provider = new GenericProvider([
            'clientId' => env('DEBTOR_MAIL_CLIENT_ID'),
            'clientSecret' => env('DEBTOR_MAIL_CLIENT_SECRET'),
            'urlAuthorize' => env('DEBTOR_MAIL_AUTH_LINK'),
            'urlAccessToken' => env('DEBTOR_MAIL_TOKEN_LINK'),
            'urlResourceOwnerDetails' => env('APP_URL')
        ]);
        $accessToken = $provider->getAccessToken('password', [
            'username' => $login,
            'password' => $password
        ]);
```

**Comment by Sammyjo20** [comment, marked as answer]
Hi,

I think what you are describing is using the “Password” OAuth2 grant. Unfortunately this is not supported in Saloon just yet. The only OAuth2 grant supported is the “Authorization Code” grant. 

You can do this in passport but it requires the user to have access to their browser.

**Reply by Sammyjo20** [reply]
However this is something I would be happy to implement after the release of v2

**Reply by AbrorGOF** [reply]
Thanks would be great.

**Reply by derheyne** [reply]
Would love to see the Password Grant coming natively to Saloon too!

**Reply by Sammyjo20** [reply]
Great, I have put it on the v2 roadmap once it's it's release v2.0.0

**Reply by juse-less** [reply]
Keep in mind that the password grant was deprecated a while back, because it's insecure.
So depending on where you use it, the support may be removed from that 'vendor'/service.
